### PR TITLE
GCI-1614: Add filing history type to DTO

### PR DESCRIPTION
**Pull request by VaughanJackson123**
Sonar report: http://code-analysis.platform.aws.chdev.org:9000/dashboard?id=uk.gov.companieshouse%3Achd-order-api&pullRequest=6.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java [hunk 1]
```diff
@@ -23,16 +23,11 @@
...
-	 */
-	public MissingImageDeliveriesController() {
-	}
-
 	@PostMapping("${uk.gov.companieshouse.chd.order.api.mid}")
 	public ResponseEntity<MissingImageDeliveriesDTO> createMissingImageDelivery(final @Valid @RequestBody MissingImageDeliveriesDTO midDTO,
 			HttpServletRequest request) {
-		Map<String, Object> logMap = LoggingUtils.createLoggingDataMap(COMPANY_NUMBER_LOG_KEY, "");
+		Map<String, Object> logMap = LoggingUtils.createLoggingDataMap(COMPANY_NUMBER_LOG_KEY,
+				midDTO.getCompanyNumber());
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java:30]: This shouldn't have tabs in it, it should use spaces instead. If the entire class is like this, we should raise the formatting as a separate Jira issue. If not, this should be re-formatted.

File: src/main/java/uk/gov/companieshouse/chd/order/api/dto/MissingImageDeliveriesDTO.java [hunk 2]
```diff
@@ -88,6 +92,14 @@ public void setPaymentReference(String paymentReference) {
...
 	}
 
+	public String getFilingHistoryType() {
+		return filingHistoryType;
+	}
+
+	public void setFilingHistoryType(String filingHistoryType) {
+		this.filingHistoryType = filingHistoryType;
+	}
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/chd/order/api/dto/MissingImageDeliveriesDTO.java:102]: This shouldn't have tabs in it, it should use spaces instead. If the entire class is like this, we should raise the formatting as a separate Jira issue. If not, this should be re-formatted.

File: src/main/java/uk/gov/companieshouse/chd/order/api/dto/MissingImageDeliveriesDTO.java [hunk 3]
```diff
@@ -32,6 +32,10 @@
 	@JsonProperty("payment_reference")
 	private String paymentReference;
 
+	@NotNull
+	@JsonProperty("filing_history_type")
+	private String filingHistoryType;
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/chd/order/api/dto/MissingImageDeliveriesDTO.java:38]: This shouldn't have tabs in it, it should use spaces instead. If the entire class is like this, we should raise the formatting as a separate Jira issue. If not, this should be re-formatted.

File: src/test/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesControllerTest.java [hunk 4]
```diff
@@ -34,6 +34,7 @@
 		MISSING_IMAGE_DELIVERIES_DTO = new MissingImageDeliveriesDTO();
 		MISSING_IMAGE_DELIVERIES_DTO.setCompanyName("Test");
 		MISSING_IMAGE_DELIVERIES_DTO.setCompanyNumber("123");
+		MISSING_IMAGE_DELIVERIES_DTO.setFilingHistoryType("TestType");
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesControllerTest.java:37]: This shouldn't have tabs in it, it should use spaces instead. If the entire class is like this, we should raise the formatting as a separate Jira issue. If not, this should be re-formatted.

File: src/test/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesControllerTest.java [hunk 5]
```diff
@@ -52,6 +53,7 @@ public void createMissingImageDeliverySuccessfully() {
 		assertThat(response.getStatusCode(), is(HttpStatus.CREATED));
 		assertEquals(MISSING_IMAGE_DELIVERIES_DTO.getCompanyName(), response.getBody().getCompanyName());
 		assertEquals(MISSING_IMAGE_DELIVERIES_DTO.getCompanyNumber(), response.getBody().getCompanyNumber());
+		assertEquals(MISSING_IMAGE_DELIVERIES_DTO.getFilingHistoryType(), response.getBody().getFilingHistoryType());
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesControllerTest.java:56]: This shouldn't have tabs in it, it should use spaces instead. If the entire class is like this, we should raise the formatting as a separate Jira issue. If not, this should be re-formatted.

File: src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java (hunk 1 above)
Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java:30]: Arghh! Caught out by this again! I'll see whether this affects whole files or just the bit I have changed. 👍

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java:30]: I'm afraid this issue seems to affect entire files. I have raised a JIRA https://companieshouse.atlassian.net/browse/GCI-1617 to deal with the formatting in a separate PR.

**Review by nabeelamirch — APPROVED** [review]

**Review by hwilliams1 — APPROVED** [review]
As the tabs to spaces formatting will be dealt with under another Jira ticket, happy to approve this one

**Pull request merged**